base/loading.jl

```diff
@@ -2382,6 +2496,15 @@ get_compiletime_preferences(uuid::UUID) = collect(get(Vector{String}, COMPILETIM
 get_compiletime_preferences(m::Module) = get_compiletime_preferences(PkgId(m).uuid)
 get_compiletime_preferences(::Nothing) = String[]
 
+function check_clone_targets(clone_targets)
+    try
+        ccall(:jl_check_pkgimage_clones, Cvoid, (Ptr{Cchar},), clone_targets)
+        return true
+    catch
+        return false
+    end
```

Comment:
why are we using try/catch for normal control flow?

Reply:
The way check_clones works is by throwing an exception, and I understand to little of the code to effectively refactor it.